### ISSUE TITLE
info / ProgressBar for iterations

### ISSUE THREAD

**Issue by aTrotier** [JuliaImageRecon/RegularizedLeastSquares.jl#71]
Following discussion in : in https://github.com/MagneticResonanceImaging/MRIReco.jl/issues/177

How can I add some way (progressBar or print) for the number of iterations and maybe other useful informations ?

I think we can pass a vector of callback to the solver, can we use that ?

**Comment by nHackel** [comment]
Yes, this can be achieved with the callbacks. The callbacks are invoked with the current solver and the current iteration, so you could just print the iteration.

By inspecting the solver one could also look at the `iterations` or look into the `iterations` parameter of the MPIReco dict to do a progress bar or X/Y done printing:

```julia
solution = solve!(S, b, callbacks = (s, i) -> println("$i/$(s.iterations)"))
```
